rego/rego.go

```diff
@@ -545,24 +695,196 @@ func (r *Rego) Compile(ctx context.Context) (*CompileResult, error) {
 	return result, nil
 }
 
-func (r *Rego) parse() (map[string]*ast.Module, ast.Body, error) {
+// PrepareFlag defines an enum type to control the behavior
+// of the Prepare() call.
+type PrepareFlag int
```

Comment:
Let's make this a bit more flexible with either an optional param struct or functional options like we use elsewhere. I'm just thinking down the road in case we need more complex parameterization.

Reply:
Can do, update coming soon.

Reply:
I updated it to the function varargs like the other stuff in here (should be familiar for users of this API).